BioForm/api/views.py

```diff
@@ -0,0 +1,143 @@
+        'date_of_birth': user.date_of_birth,
+    }
+
+
+class RegisterView(APIView):
+    def post(self, request):
+        serializer = RegisterSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data)
+
+
+class LoginView(APIView):
+    def post(self, request):
+        email = request.data['email']
+        password = request.data['password']
+
+        user = Account.objects.filter(email=email).first()
+
+        if user is None:
+            raise AuthenticationFailed('User not found!')
+
+        if not user.check_password(password):
+            raise AuthenticationFailed('Incorrect Password!')
+
+        payload = {
+            'id': user.id,
+            'exp': datetime.datetime.utcnow() + datetime.timedelta(minutes=60),
+            'iat': datetime.datetime.utcnow()
+        }
+
+        token = jwt.encode(payload, 'secret', algorithm='HS256').decode('utf-8')
+
+        response = Response()
+        response.set_cookie(key='jwt', value=token, httponly=True)
+        response.data = {
+            'jwt': token,
+        }
+
+        return response
+
+
+class LogoutView(APIView):
+    def post(self, request):
+        response = Response()
+        response.delete_cookie('jwt')
+        response.data = {
+            'message': 'success',
+        }
+        return response
+
+
+class BioView(APIView):
+
+    permission_classes = [IsJWTAuthenticated]
+
+    def get(self, request):
+        try:
+            serializer = request.api_user
+            bio = Bio.objects.filter(user_id=serializer.data['id']).first()
+            bio_serializer = BioSerializer(bio)
+            return Response(bio_serializer.data)
+        except:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+
+
+    def post(self, request):
+        serializer = request.api_user
+
+        if Bio.objects.filter(user_id=serializer.data['id']):
+            return Response(
+                    {'detail': 'Bio already exists!',
+                    'status': '400'}, 
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+
+        bio_serializer = BioSerializer(data=request.data)
+        bio_serializer.initial_data['user'] = serializer.data['id']
+        bio_serializer.is_valid(raise_exception=True)
+        bio_serializer.save()
+
+        return Response({
+            'message': 'success',
+            'record_entered': bio_serializer.data
+        })
+
+
+    def put(self, request):
+
+        try:
+            serializer = request.api_user
+            bio = Bio.objects.filter(user_id=serializer.data['id']).first()
+
+            bio_serializer = BioSerializer(instance=bio, data=request.data, partial=True)
+
+            if bio_serializer.is_valid():
+                bio_serializer.save()
+                return Response(bio_serializer.data)
+            else:
+                return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
If provided data is not valid server should return 400 bad request error not 404 not found error.